third_party_plugin_dev_portal/VST3presetSwitchingPerformEditAlternatives.md

```diff
@@ -0,0 +1,199 @@
+---
+layout: link
+active_page: VST3presetSwitchingPerformEditAlternatives.
+title: Recommendations to VST3 plugins manufacturers regarding bulk calls to performEdit
+---
+
+# Recommendations to VST3 plugins manufacturers regarding bulk calls to performEdit
+## Introduction
+
+For some VST3 plugins, Ableton receives reports of user interface lag when users switch presets from the plugin editor. After investigation, it seems that some VST3 manufacturers implement preset changes from the VST3 editor by calling performEdit for all parameters in order to notify both the DAW and the VST3 controller counterpart of its new state.
+
+Live needs time to react to performEdit in order for example to provide shadow-mapping of unmapped parameters, undo-redo.
```

Comment:
It's not so clear what "shadow mapping of unmapped parameters" means. Maybe put "for our parameter mapping system"?

Reply:
I agree that "shadow mapping of unmapped parameters" is not a VST3 feature, nor a widely known feature of Live.

```suggestion
Live needs time to react to performEdit in order for our parameter mapping system.
```